addons/gradein/models/gradein_order.py

```diff
@@ -45,21 +46,49 @@ class GradeInOrder(models.Model):
     partner_id = fields.Many2one(
         comodel_name="res.partner", string="Cliente", required=True
     )
-    price = fields.Float(
-        string="Importe a pagar", help="Price that client will pay", required=True
+
+    price = fields.Monetary(
+        string="Importe a pagar",
+        currency_field="currency_id",
+        help="Price that client will pay",
+        required=True,
     )
-    question_answer_id = fields.One2many(
+
+    currency_id = fields.Many2one(related="equipment_id.currency_id")
+    question_answer_ids = fields.One2many(
         comodel_name="gradein.question.answer",
         inverse_name="order_id",
         string="Respuestas",
         required=True,
     )
 
-
     @api.model
     def create(self, vals_list):
-        if vals_list.get('name', ('New')) == ('New'):
-            vals_list['name'] = self.env['ir.sequence'].next_by_code(
-                'gradein.order.name') or ('New')
+        if vals_list.get("name", ("Nueva Orden")) == ("Nueva Orden"):
+            vals_list["name"] = self.env["ir.sequence"].next_by_code(
+                "gradein.order.name"
+            ) or ("Nueva Orden")
         res = super().create(vals_list)
-        return res
\ No newline at end of file
+        return res
+
+    @api.onchange("equipment_id")
+    def on_change_equipment(self):
+        """When you select the team you get the questions and answers from the form"""
+        lines = [(5, 0, 0)]
+        if self.equipment_id:
+            for question in self.equipment_id.equipment_type_id.question_ids:
+                questions_dict = {"question_id": question.id}
+                lines.append((0, 0, questions_dict))
+        self.question_answer_ids = lines
```

Comment:
```suggestion
    @api.onchange("equipment_id")
    def on_change_equipment(self):
        """When you select the equipment you get the questions and answers from the form"""
        commands_data = [(5, 0, 0)]  # We delete all the questions first
        if self.equipment_id:
            for question in self.equipment_id.equipment_type_id.question_ids:
                questions_dict = {"question_id": question.id}
                commands_data.append((0, 0, questions_dict))  # We create this records with the questions of the equipment
        self.question_answer_ids = commands_data
```

Es una sugerencia para que se entienda mejor el codigo,  lo que hacen esos numeros. Renombre la variable `lines` a algo que tengas mas sentido de lo que guarda, sentite libre a renombrarla como mejor te parezca.